### PR TITLE
Make dependency of vissl optional

### PR DESCRIPTION
**Pull request by ar90n**
## What does this PR do?
This PR is motivated to achieve the second change in [this comment](https://github.com/PyTorchLightning/lightning-flash/issues/1031#issuecomment-1083103525). And this PR contains the following modifications.

* Make `training_strategy`, `head`, `pretraining_transform` in the arguments of `__init__` of ImageEmbedder optional
* Add `DefaultAdapter` to the Embedding task which is inspired by the Image Classification task's one.
* Fix a test about integration with Fiftyone to work correctly
* Add test cases for only embedding without VISSL

Part of https://github.com/PyTorchLightning/lightning-flash/issues/1031

This PR is based on #1271. Because its target branch was wrong.
I'm sorry for my wrong operations. I misunderstood that it would be okay to change its target branch.

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? [not needed for typos/docs]
- [x] Did you verify **new and existing tests pass** locally with your changes?
- [x] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 
+- Change the dependency to VISSL to optional ([#1276](https://github.com/PyTorchLightning/lightning-flash/pull/1276))
```
Comment (ethanwharris) [inline CHANGELOG.md:19]:
```suggestion
- Changed the `ImageEmbedder` dependency on VISSL to optional ([#1276](https://github.com/PyTorchLightning/lightning-flash/pull/1276))
```

File: tests/image/embedding/test_model.py [hunk 2]
```diff
@@ -87,3 +87,31 @@ def test_vissl_training(backbone, training_strategy, head, pretraining_transform
...
+        batch_size=4,
+        transform_kwargs=dict(image_size=(224, 224)),
+    )
+
+    embedder = ImageEmbedder(backbone=backbone)
+
+    trainer = flash.Trainer(
+        gpus=torch.cuda.device_count(),
+        detect_anomaly=True,
+    )
```
Comment (ethanwharris) [inline tests/image/embedding/test_model.py:112]: Shouldn't need the `detect_anomaly`:
```suggestion
    trainer = flash.Trainer(
        gpus=torch.cuda.device_count(),
    )
```

File: flash/image/embedding/strategies/default.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        backbone: torch.nn.Module,
+        **kwargs,
+    ) -> Adapter:
+        adapter = cls(backbone)
+        adapter.__dict__["_task"] = task
+        return adapter
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        raise NotImplementedError(
+            "Training a `ImageEmbedder` with `DefaultAdapter` is not supported. Use a different strategy instead."
```
Comment (ethanwharris) [inline flash/image/embedding/strategies/default.py:51]:
```suggestion
            "Training an `ImageEmbedder` with `strategy="default"` is not supported. Use a different strategy instead."
```

File: flash/image/embedding/strategies/default.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        )
+
+    def validation_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DataKeys.INPUT], batch[DataKeys.TARGET])
```
Comment (ethanwharris) [inline flash/image/embedding/strategies/default.py:55]: Let's raise an error for validation and test too :smiley:

**Review by ethanwharris — COMMENTED** [review]
Awesome :smiley: Just a couple of minor comments

File: flash/image/embedding/model.py [hunk 5]
```diff
@@ -76,9 +76,9 @@ class ImageEmbedder(AdapterTask):
 
     def __init__(
         self,
-        training_strategy: str,
-        head: str,
-        pretraining_transform: str,
+        training_strategy: str = "default",
```
Comment (ethanwharris) [inline flash/image/embedding/model.py:79]: can also change `required_extras` for this task to be:
```
required_extras: str = "image"
```

**Comment by ar90n** [comment]
This PR is ready for review. Please review it.
I have one question about dealing with DeepSource. Could I ignore its failure as same as other PRs?

I'm sorry for making a messy commit log. I struggled to solve CI issues.

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley: @ar90n Fine to ignore the deep source issues, we plan to get rid of it. No worries about the commit history! That's why we always squash and merge 😆

**Comment by ar90n** [comment]
@ethanwharris 
Thanks for your review. I'm so glad to make this PR accepted. I'm looking forward to this PR being merged.

**Comment by ethanwharris** [comment]
@ar90n No wories, thanks for your work on this!! Very glad to see it all working smoothly 😃

**Pull request merged**